operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -59,9 +59,23 @@ var (
 	testDataDir = filepath.Join(operatorRootDir, "cmd/mesh/testdata/manifest-generate")
 
 	// Snapshot charts are in testdata/manifest-generate/data-snapshot
-	snapshotCharts = chartSourceType(filepath.Join(testDataDir, "data-snapshot"))
+	snapshotCharts = func() chartSourceType {
+		d, err := os.MkdirTemp("", "data-snapshot-")
+		if err != nil {
+			panic(fmt.Errorf("failed to make temp dir: %v", err))
+		}
+		f, err := os.Open("testdata/manifest-generate/data-snapshot.tar.gz")
+		if err != nil {
+			panic(fmt.Errorf("failed to read data snapshot: %v", err))
+		}
+		if err := tgz.Extract(f, d); err != nil {
+			panic(fmt.Errorf("failed to extract data snapshot: %v", err))
+		}
+		return chartSourceType(filepath.Join(d, "manifests"))
+	}()
 	// Compiled in charts come from assets.gen.go
 	compiledInCharts chartSourceType = "COMPILED"
+	_                                = compiledInCharts
```

Comment:
if this is just to silence lint, can compiledInCharts be removed altogether at this point?

Reply:
there are no tests currently using it, I wasn't sure if that was intended or a gap